src/Kestrel.Core/KestrelServerOptions.cs

```diff
@@ -100,8 +100,7 @@ public void Listen(IPEndPoint endPoint, Action<ListenOptions> configure)
                 throw new ArgumentNullException(nameof(configure));
             }
 
-            var listenOptions = new ListenOptions(endPoint) { KestrelServerOptions = this };
-            configure(listenOptions);
+            var listenOptions = new ListenOptions(endPoint) { KestrelServerOptions = this, Configure = configure };
```

Comment:
Instead of adding this weird Configure callback, why not keep it the way it was and invoke configure directly in Listen()? Is this just for testing? Seems super confusing to me.